packages/cli/src/commands/dev.ts

```diff
@@ -143,7 +144,45 @@ type LocalTunnelResponse = {
   url: string;
 };
 
-async function createTunnel(localOrigin: string) {
+async function tunnelHealthCheck(configTunnelUrl: string) {
+  try {
+    const res = await (
+      await fetch(configTunnelUrl + '/api/novu?action=health-check', {
+        method: 'GET',
+        headers: {
+          accept: 'application/json',
+          'Content-Type': 'application/json',
+        },
+      })
+    ).json();
+
+    return res.status === 'ok';
+  } catch (e) {
+    return false;
+  }
+}
+
+async function createTunnel(localOrigin: string): Promise<string> {
+  const delay = (ms: number) => new Promise((resolve) => setTimeout(resolve, ms));
+
+  const configTunnelUrl = config.getValue('tunnelUrl');
+  const storeUrl = configTunnelUrl ? new URL(configTunnelUrl) : null;
+  const originUrl = new URL(localOrigin);
+
+  if (storeUrl) {
+    await connectToTunnel(storeUrl, originUrl);
+    await delay(100);
```

Comment:
there is a race with the time it takes to establish a connection with the socket.
initially, I created a retry mechanism but in the end, i refactored to 100ms await to simplify the flow.